From 054a8e0787b70cc04a1360b5a42761f44cf926d6 Mon Sep 17 00:00:00 2001
From: Peter Chen <peter.chen@freescale.com>
Date: Sun, 15 Feb 2015 15:44:43 +0800
Subject: [PATCH 1376/1594] MLK-10279-2 usb: chipidea: imx: only do charger
 detection if we enable it

commit 0f55f2d1c59532c516a8b3878f0d89f2b1ea717a from
git://git.freescale.com/imx/linux-2.6-imx.git

We should only do charger detection if both we have enabled
usb charger at kernel configuration and set usb charger enable
at dts. Otherwise, we will do useless things, like toggle dp.

Signed-off-by: Peter Chen <peter.chen@freescale.com>
---
 drivers/usb/chipidea/ci_hdrc_imx.c |    9 +++++----
 1 files changed, 5 insertions(+), 4 deletions(-)

diff --git a/drivers/usb/chipidea/ci_hdrc_imx.c b/drivers/usb/chipidea/ci_hdrc_imx.c
index b10c39c..0a7f380 100644
--- a/drivers/usb/chipidea/ci_hdrc_imx.c
+++ b/drivers/usb/chipidea/ci_hdrc_imx.c
@@ -303,11 +303,12 @@ static int ci_hdrc_imx_notify_event(struct ci_hdrc *ci, unsigned event)
 	switch (event) {
 	case CI_HDRC_CONTROLLER_VBUS_EVENT:
 		if (data->usbmisc_data && ci->vbus_active) {
-			if (data->imx6_usb_charger_detection)
+			if (data->imx6_usb_charger_detection) {
 				ret = imx6_usb_vbus_connect(&data->charger);
-			if (!ret && data->charger.psy_desc.type !=
-						POWER_SUPPLY_TYPE_USB)
-				ret = CI_HDRC_NOTIFY_RET_DEFER_EVENT;
+				if (!ret && data->charger.psy_desc.type !=
+							POWER_SUPPLY_TYPE_USB)
+					ret = CI_HDRC_NOTIFY_RET_DEFER_EVENT;
+			}
 		} else if (data->usbmisc_data && !ci->vbus_active) {
 			if (data->imx6_usb_charger_detection)
 				ret = imx6_usb_vbus_disconnect(&data->charger);
-- 
1.7.5.4

